internal/sqlutil/sql.go

```diff
@@ -38,9 +38,18 @@ type Transaction interface {
 // was applied correctly. For example, 'database is locked' errors in sqlite will happen here.
 func EndTransaction(txn Transaction, succeeded *bool) error {
 	if *succeeded {
-		return txn.Commit() // nolint: errcheck
+		return txn.Commit()
 	} else {
-		return txn.Rollback() // nolint: errcheck
+		return txn.Rollback()
+	}
+}
+
+// CheckEndTransactionErr ends a transaction and overwrites the error pointer if its value was nil.
+// If the transaction succeeded then it is committed, otherwise it is rolledback.
+// Designed to be used with defer (see EndTransaction otherwise).
+func CheckEndTransactionErr(txn Transaction, succeeded *bool, err *error) {
```

Comment:
Slighty confusing naming: the `Check` implies that you've already called `EndTransaction` and are now checking the result, but we're actually calling `EndTransaction` here. `EndTransactionAndCheck` would be clearer but still awkward.

Reply:
What do you think of `EndTransactionWithCheck`? (`With` is quite usual in the standard library)

Reply:
Sounds good to me.